applications/OptimizationApplication/python_scripts/algorithms/algorithm.py

```diff
@@ -68,3 +69,8 @@ def CallOnAllProcesses(self, process_types: 'list[str]', *args, **kwargs):
         for process_type in process_types:
             CallOnAll(self._optimization_problem.GetListOfProcesses(process_type), *args, **kwargs)
 
+    def _InitializeIteration(self) -> None:
+        list(map(lambda x: x.ExecuteInitializeSolutionStep(), self._optimization_problem.GetAvailableProcessTypes()))
```

Comment:
you sure you need `list(...)`? you are not returning it
or is that bcs of the generator?

Reply:
AFAIK, the `map` functions are not evaluated unless they are used. Thats the reason.